task19/js/slider.js

```diff
@@ -0,0 +1,163 @@
+import '../css/style.css'
+import { createElementWithClassName } from "./utils/createElement";
+import { addLoading } from "./modules/loading";
+import { removeLoading } from "./modules/loading";
+import { renderErrorMessage } from "./modules/error";
+
+const endpointForSlider = "https://mocki.io/v1/fc890b98-dee0-4eb6-9e83-8c25364dcaa9";
+
+const slideShowContainer = document.getElementById("js-slideShowContainer");
+const slideShowWrapper = document.getElementById("js-slideShowWrapper");
+const slideImageList = document.getElementById("js-slideImageList");
+let currentImageIndex = 0;
+
+const getFetchData = async () => {
+  const response = await fetch(endpointForSlider);
+  if (!response.ok) {
+    console.error(`${response.status}:${response.statusText}`);
+    renderErrorMessage(slideShowWrapper, "問題が発生し表示することができません。");
+  }
+  const json = await response.json();
+  return json;
+};
+
+const fetchImageData = () => {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(getFetchData(endpointForSlider));
+    }, 3000);
+  });
+};
+
+const initialize = async () => {
+  addLoading(slideShowWrapper);
+  let slideImageData;
+  try {
+    slideImageData = await fetchImageData();
+  } catch (error) {
+    console.error(error.message);
+    renderErrorMessage("表示することができません");
```

Comment:
IMO
I think you are missing an argument to pass to `renderErrorMessage`.
``` js
export const renderErrorMessage = (parent, errorMessage) => {
    parent.textContent = errorMessage;
  };
```

Reply:
Thank you for pointing this out. I had forgotten about it.
[3fa83df](https://github.com/seeeen044/JavaScript_handson/pull/20/commits/3fa83dff15ef174382a576826f6d11817b53af56)